### PR TITLE
fix(core): prevent multiple call `markForCheck` in dropdown

### LINKED ISSUE #997: 🐞 - Sub dropdown after close gets additionally reopened and closed in left top corner
### Which @taiga-ui/* package(s) are the source of the bug?

Don't known / other

### Please provide a link to a minimal reproduction of the bug

_No response_

### Description

Steps to reproduce:
- run taiga demo app locally
- change this file `projects/demo/src/modules/components/data-list/examples/2/index.ts`
```ts
selectOption(item: string) {
        this.dropdownOpen = false;
        this.dialogService.open(`You selected ${item}`).subscribe(); // <-- comment this line
}
```
- open http://localhost:4200/components/data-list#submenu and observe next steps here

Nov-12-2021 13-21-54

### Which browsers have you used?

- [X] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### Which operating systems have you used?

- [X] macOS
- [ ] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes #997

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/core/abstract/abstract-dropdown.ts [hunk 1]
```diff
@@ -100,6 +106,15 @@ export abstract class AbstractTuiDropdown
         return checkFixedPosition(this.elementRef.nativeElement);
     }
 
+    @tuiPure
+    protected switchDropdown(value: boolean) {
```
Comment (waterplea) [inline projects/core/abstract/abstract-dropdown.ts:110]: toggleDropdown

File: projects/demo/src/modules/components/data-list/examples/2/index.html [hunk 2]
```diff
@@ -69,7 +69,7 @@
         </tui-data-list>
     </ng-template>
     <ng-template #drinksTmp>
-        <tui-data-list>
+        <tui-data-list [tuiActiveZoneParent]="activeZone">
```
Comment (waterplea) [inline projects/demo/src/modules/components/data-list/examples/2/index.html:72]: Is it really necessary? I think it is only needed if this data list has submenus.

File: projects/core/abstract/abstract-dropdown.ts [hunk 3]
```diff
@@ -67,24 +67,30 @@ export abstract class AbstractTuiDropdown
...
         private readonly injector: Injector,
         private readonly portalService: TuiPortalService,
         protected readonly elementRef: ElementRef<HTMLElement>,
         readonly activeZone: TuiActiveZoneDirective | null,
-    ) {}
+    ) {
+        /**
+         * @note: component updates itself in ngAfterViewChecked
+         */
+        this.dropdownBoxRef?.changeDetectorRef.reattach();
```
Comment (waterplea) [inline projects/core/abstract/abstract-dropdown.ts:80]: Did you mean `detach`?

Comment (splincode) [reply projects/core/abstract/abstract-dropdown.ts]: fixed

File: projects/demo/src/modules/components/data-list/examples/2/index.html (hunk 2 above)
Comment (splincode) [reply projects/demo/src/modules/components/data-list/examples/2/index.html]: fixed

File: projects/core/abstract/abstract-dropdown.ts (hunk 1 above)
Comment (splincode) [reply projects/core/abstract/abstract-dropdown.ts]: fixed

**Comment by splincode** [comment]
1-initial snap
2-opened snap
3-converted-money snap
4-email-opened snap
5-range-opened snap
6-calendar-opened snap
7-finish snap

**Comment by splincode** [comment]
@waterplea @vladimirpotekhin please, review me)

**Review by waterplea — APPROVED** [review]

**Pull request merged**